validation-test/Sema/rdar34333874.swift

```diff
@@ -0,0 +1,31 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+class C {
+  var a: Int = 0
+  var b: Int = 0
+}
+
+@inline(never)
+func foo<T>(_ item: T, update: (inout T) throws -> Void) rethrows -> T {
+  var this = item
+  try update(&this)
+  return this
+}
+
+let rdar34333874 = foo(C()) {
```

Comment:
Hmm, apparently I didn't submit my comment about the test. Are you sure this is testing the same thing as the original report? This appears to actually be mutating something whereas the original report wasn't.

Reply:
Both me and @swiftix replied to that one and I've added one more test in addition to the mutating one from SR-5861 which does mutate the fields of the argument.